File: src/main/java/ru/tisov/denis/platform/service/ContainerService.java
package ru.tisov.denis.platform.service;

import ru.tisov.denis.platform.domain.docker.Container;

import java.util.List;

public interface ContainerService {

    void restart(Container container);

    void start(Container container);

    void stop(Container container);

    void remove(Container container);

    void createAndStart(Container container);

    void create(Container container);

    List<Container> getRunning(Long hostId, Long environmentId);

    List<Container> getStopped(Long hostId, Long environmentId);
}

File: src/main/resources/static/app/js/service/container-service.js
dockerApp.service('containerService', function (restService) {

    this.delete = function (container) {
        return restService.delete("/action/container/delete/" + container.hostName + "/" + container.id);
    };

    this.start = function (container) {
        return restService.post("/action/container/start", container)
    };

    this.stop = function (container) {
        return restService.post("/action/container/stop", container)
    };

    this.restart = function (container) {
        return restService.post("/action/container/restart", container)
    };

});
